content/doc/book/pipeline/shared-libraries.adoc

```diff
@@ -700,3 +700,29 @@ Only entire ``pipeline``s can be defined in shared libraries as of this time.
 This can only be done in `vars/*.groovy`, and only in a `call` method. Only one
 Declarative Pipeline can be executed in a single build, and if you attempt to
 execute a second one, your build will fail as a result.
+
+=== Testing library pull request changes
+
+By adding `@Library('my-shared-library@pull/<your-pr-number>/head') _` at the top of the library consumer Jenkinsfile, you can test your pipeline library pull request changes _in situ_ if your Pipeline library is hosted on GitHub.
+Refer to the pull request or merge request branch naming convention for other providers like Assembla, BitBucket, Gitea, GitLab, and Tuleap.
+
+Taking for example a change to the global ci.jenkins.io shared pipeline which source code is stored at https://github.com/jenkins-infra/pipeline-library/.
```

Comment:
Note that intuitively I would have kept the other way around, as it's not the pipeline which is storing the source code.

Reply:
The other idea I had for this was 

Take, for example, a change to the global ci.jenkins.io shared pipeline, whose source code is stored at https://github.com/jenkins-infra/pipeline-library/.

My hesitation there was using "whose" since it is not a person, but I was trying to not misconstrue the information. If my suggestion invalidates the message, please disregard.

Reply:
"Which" can't be used in that case?

Reply:
According to https://english.stackexchange.com/questions/23541/can-whose-refer-to-an-inanimate-object "whose" seems fine.

Reply:
WDYT of

> Take, for example, a change to the global ci.jenkins.io shared pipeline for which the source code is stored at https://github.com/jenkins-infra/pipeline-library/.

Reply:
What about 

"Take, for example, a change to the global ci.jenkins.io shared pipeline, which has source code stored at https://github.com/jenkins-infra/pipeline-library/."

There just needs to be something for the "which" to connect to, otherwise it doesn't necessarily make sense. I can see how my first suggestion would imply that this action is storing the source code, and how that could be incorrect though, so I'm wondering if this might be better/more accurate?